sentry_streams/src/consumer.rs

```diff
@@ -345,7 +357,8 @@ mod tests {
 
             let submitted_messages = Arc::new(Mutex::new(Vec::new()));
             let submitted_messages_clone = submitted_messages.clone();
-            let next_step = FakeStrategy::new(submitted_messages, false);
+            let submitted_watermarks = Arc::new(Mutex::new(Vec::new()));
+            let next_step = FakeStrategy::new(submitted_messages, submitted_watermarks, false);
```

Comment:
Please consider passing a watermark mesage in this test and validate it ends up in `submitted_watermarks`